다 합치고 pr 날립니다

april0114 opened the pull request:
- 추후 수정 (json 바디 요청 형식 변경과 log 추가)

april0114 committed:
♻️ Refactor : topthree 삭제

- 통신 이후 거북목 top3인 컨트롤러, repository, service 삭제

april0114 committed:
️️♻️ Refactor : 요청받는 json형식 변환

- user의 정보에서 다 받아오는 것에서 -> nickname과 score만 받아오는 걸로 변환

april0114 committed:
🐛 Fix : json바디 형식 수정

- 실행하려면 security config 건드려야함
- user를 받아오지 않고 login_id만 저장함

april0114 committed:
✨ Feat : 챌린지 개인 저장 기능

- 챌린지 id로 제목을 가져와서 개인이 참여중인 챌린지 저장
-get으로 개인이 참여중인 챌린지 조회 가능

april0114 committed:
✨ Feat : 오늘의 운동 기록 저장 구현

- json 바디 형식 변경 필요

hyuneu-n committed:
✨ Feat: 운동 완료 시 포인트 자동 지급 기능 추가

hyuneu-n committed:
✨ Feat: 메인 뱃지 설정 기능 추가

april0114 committed:
🐛 Fix : securityconfig 수정

april0114 committed:
Merge pull request #40 from nowjiin/feature/new_exercise

🐛 Fix : json바디 형식 수정

april0114 committed:
Merge pull request #41 from nowjiin/feature/new_turtle

Feature/new turtle

april0114 committed:
Merge pull request #42 from nowjiin/feature/new_challenge

✨ Feat : 챌린지 개인 저장 기능

april0114 committed:
Merge pull request #44 from nowjiin/feature/new_point

✨ Feat: 운동 완료 시 포인트 자동 지급 기능 추가

april0114 committed:
Merge pull request #45 from nowjiin/feature/new_badge

✨ Feat: 메인 뱃지 설정 기능 추가

april0114 committed:
🐛 Fix: security config

april0114 committed:
Merge pull request #43 from nowjiin/feature/new_today_exercise

✨ Feat : 오늘의 운동 기록 저장 구현

april0114 committed:
🐛 Fix: "mypage" securityconfig 추가

april0114 committed:
Merge pull request #46 from nowjiin/feature/security_config

🐛 Fix: "mypage" securityconfig 추가

nowjiin commented on src/main/java/com/hotspot/livfit/badge/controller/BadgeController.java:
![image](https://github.com/user-attachments/assets/210a3d69-cab2-4f8b-8831-6df63586e28c)

### /api/userbadges/mybadge
- 내 보유 중인 뱃지를 요청할 경우 중복 된 뱃지가 있어요. 이부분을 어떻게 처리할 건지 고민이 필요해보입니다.
- login 할때 뱃지를 바로 부여 하겠다고 가정을 해볼께요. 그럼 로그인 뱃지가 로그인 시 마다 계속 쌓일 것 같습니다

nowjiin commented on src/main/java/com/hotspot/livfit/today_exercise/controller/TodayExerciseController.java:
### /api/today_exercise/save

- POST 요청시
```
{
  "dayOfWeek": "MONDAY",
  "success": "true"
}
```

응답은 이렇게 잘와요
![image](https://github.com/user-attachments/assets/d2ffc1f5-2cce-4be0-8d3d-2cb8de938ef4)

하지만 DB에 

![image](https://github.com/user-attachments/assets/68279b3c-9c86-4aeb-ad01-5c97a30f7947)

위와 같이 day_of_week가 저장이 안됩니다.

추가로 비빌번호 토큰 유져role 빼주세요

nowjiin replied:
위와 같이 day_of_week가 저장이 안됩니다.

=> DB에 저장이 안된다는 뜻입니다 !

nowjiin replied:
![image](https://github.com/user-attachments/assets/3ef7eb38-01f8-4498-b7ac-b7ba29422086)

Authorization 에 Bearer를 받고있는데
FE단에서 loginId를 보낼 수 있는 방법이 없어보입니다

RequestBody에 login Id를 제거해주세요

![image](https://github.com/user-attachments/assets/f888b814-af1c-47fc-9113-1daedb9dbd9c)

이렇게 요청을 해도 응답이 잘옵니다 
제거 해야할 것 같아요

nowjiin commented on src/main/java/com/hotspot/livfit/exercise/controller/SquatController.java:
102번째 줄 sout 빼주세요

nowjiin replied:
![image](https://github.com/user-attachments/assets/16a60087-9414-476b-98f9-fba7608e9a69)

이런식으로 콘솔에 찍힙니다 서버에서도요

nowjiin commented on src/main/java/com/hotspot/livfit/mypage/service/MyPageService.java:
### /api/mypage

컨트롤러가 없어 여기에 리뷰 남깁니다
![image](https://github.com/user-attachments/assets/846f1dac-7f6a-4f7f-9ef9-3d6ecaeff1b4)

비번 토큰 role 응답에서 빼주세요

nowjiin replied:
![image](https://github.com/user-attachments/assets/76545439-2c06-43a0-a910-2d71bf2d7fe1)

컨트롤러에 대한 리뷰를 여기에 남길께요 !
```
@ApiResponses({
    @ApiResponse(responseCode = "200", description = "가져오기 완료."),
    @ApiResponse(responseCode = "400", description = "잘못된 요청."),
    @ApiResponse(responseCode = "500", description = "서버 에러.")
  })
```

이게 컨트롤러에 붙어있는거같아요 지워주세요!

nowjiin commented on src/main/java/com/hotspot/livfit/point/controller/PointController.java:
### 
- /api/points
![image](https://github.com/user-attachments/assets/4b461765-e22f-4022-a917-a4136a7b9460)

응답에 토탈 포인트 + points도 같이 응답으로 줘야 할 것 같아요!
이전에 쓴 포인트 혹은 얻은 포인트도 보여줘야 할 것같아서 !

![image](https://github.com/user-attachments/assets/0546c431-569d-482d-870e-a3c3c4556133)

nowjiin commented on src/main/java/com/hotspot/livfit/config/SecurityConfig.java:
api/turtle/**"

-> /api/turtle/**"로 변경 요청

    "/api/mypage/**",
앞에 탭 제거 요청

nowjiin replied:
api/turtle/**
을
/api/turtle/** 로 바꿔주세요

hyuneu-n committed:
♻️ Refactor: 뱃지 중복 부여 못하도록 설정

hyuneu-n committed:
♻️ Refactor: 뱃지 중복 부여 못하도록 설정 및 마이페이지 수정

마이페이지에 추가한 뱃지개수 get 로직이 유저뱃지레포지토리에 추가한 쿼리를 가져다써서 어쩔 수 없이 뱃지 브랜치에 한번에 푸쉬합니다

hyuneu-n committed:
♻️ Refactor: 포인트 차감/적립 값 request에 추가

hyuneu-n committed:
♻️ Refactor: mypage에서 정보get해오는거 엔티티에서 DTO로 수정

마이페이지에 추가한 뱃지개수 get 로직이 유저뱃지레포지토리에 추가한 쿼리를 가져다써서 어쩔 수 없이 뱃지 브랜치에 한번에 푸쉬합니다.

april0114 committed:
♻️ Refactor : sout 삭제

- lunge, controller, squat 다 sout 삭제했습니다

april0114 committed:
🐛 Fix : Json 바디, 스트링 타입 변환

- JSON 바디 loginid만 받아오는 것으로 변경
- dayoffweek -> 0 에서 스트링 타입으로 변경해 monday 저장되는것 확인

april0114 committed:
♻️ Refactor : 각 컨트롤러에 log 표시

- 기록 저장 시
- 기록 조회시
- 그래프 기록 조회 시

hyuneu-n committed:
♻️ Refactor: point 로그 추가

hyuneu-n committed:
♻️ Refactor : 뱃지 log 추가

april0114 committed:
🐛 Fix: : JSON 바디 형식 수정

- 데이터 조회 수정 필요

april0114 committed:
Merge pull request #50 from nowjiin/feature/refactor-exercise

♻️ Refactor : sout 삭제

april0114 committed:
Merge pull request #51 from nowjiin/feature/refactor-badge

♻️ Refactor: 뱃지 및 마이페이지 수정

april0114 committed:
Merge pull request #52 from nowjiin/feature/refactor-point

♻️ Refactor: 포인트 차감/적립 값 request에 추가  및 로그 추가

april0114 committed:
Merge pull request #53 from nowjiin/feature/refactor-challenge

🐛 Fix: : JSON 바디 형식 수정

april0114 committed:
Merge pull request #54 from nowjiin/feature/refactor-today_exercies

🐛 Fix : Json 바디, 스트링 타입 변환

hyuneu-n commented:
1. 뱃지 존재 여부 확인하는 로직 추가하여 중복 부여 못하도록 설정
![353518248-8526f097-70f8-49f7-b331-6690bb430600](https://github.com/user-attachments/assets/d525950a-a16c-4f94-973a-d52c947e3eb2)

2. 마이페이지에 추가한 뱃지개수 get 로직이 유저뱃지레포지토리에 추가한 쿼리를 가져다써서 어쩔 수 없이 뱃지 브랜치에 한번에 푸쉬함

3. 뱃지 개수는 피그마 보니까 마이페이지에 있어서 추가함

4. mypage에서 정보get해오는거 엔티티에서 DTO로 수정 및 컨트롤러 ApiResponses 삭제

5. 로그 간단하게 추가

6. 포인트 차감/적립값 response에 추가

nowjiin commented on src/main/java/com/hotspot/livfit/badge/service/UserBadgeService.java:
안쓰는줄 주석 날려주세요

nowjiin commented on src/main/java/com/hotspot/livfit/challenge/controller/ChallengeController.java:
private static final Logger logger = LoggerFactory.getLogger(ChallengeController.class);
를 
@sl4fj 로 바꿔 주세요

nowjiin commented on src/main/java/com/hotspot/livfit/exercise/controller/LungeController.java:
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

이거말고 어노테이션으로 대체해주세요

@slf4j 어노테이션 맨앞에 붙이면되요

jwtUtils 참고
![image](https://github.com/user-attachments/assets/330c7eab-894e-499b-a919-f043ba8b2f2f)

import문은
import lombok.extern.slf4j.Slf4j;
이거 import 하고 쓰면 됩니다

사소하긴해요
롬복을 사용중이니까 롬복의 slf4j를 쓰는느낌

nowjiin commented on src/main/java/com/hotspot/livfit/exercise/controller/PushupController.java:
위와 마찬가지로 

log를 롬복의 slf4j로 써주세여

nowjiin commented on src/main/java/com/hotspot/livfit/mypage/controller/MyPageController.java:
현재 /api/mypage 로 get요청시

Lunge, Pushup, Squat의 운동 정보를 가져와요

하지만 Figma의 마이페이지를 보면

이 운동은 가져오지 않아요

필요없는 데이터를 mypage에서 이렇게 가져올 필요가 없어보입니다

활용하지 않는 로직은 아니고 나중에 운동 기록 api를 만들때 넣어야 할 것 같습니다 

추후 운동 기록 api를 생성할 경우 이부분을 뺴주세요
